src/v/coproc/wasm_event.cc

```diff
@@ -107,30 +127,42 @@ wasm::errc validate_event(const model::record& r) {
     if (auto errc = std::get_if<wasm::errc>(&action)) {
         return *errc;
     }
+
+    auto get_type_res = get_coproc_type(r);
+    if (auto err = std::get_if<wasm::errc>(&get_type_res)) {
+        return wasm::errc::unexpected_coproc_type;
+    }
+    header.type = std::get<event_type>(get_type_res);
```

Comment:
can you use `ss::visit` here so that we can get the compiler to type in case more types are added to the variant?

Reply:
done